.github/workflows/ondemand-main.yaml

```
@@ -0,0 +1,142 @@
+name: Ondemand build
+
+on:
+  workflow_dispatch:
+    inputs:
+      branch_name:
+        description: 'The name of the branch to build.'
+        default: 'maintenance-3.0.x'
+        required: true
+      sauce_labs:
+        description: 'Should unit tests be run on Sauce Labs?'
```

We should probably annotate in the description what we expect as value. Examples:
```suggestion
        description: 'Should unit tests be run on Sauce Labs? (set to empty to disable this)'
```
or
```suggestion
        description: 'Should unit tests be run on Sauce Labs? (leave empty for "no")'
```